apps/vc-api/src/key/key.service.ts

```diff
@@ -85,32 +82,66 @@ export class KeyService implements IGenerateKey {
     if (key.privateKey.crv !== 'Ed25519' || key.publicKey.crv !== 'Ed25519') {
       throw new BadRequestException('Only Ed25519 keys are supported');
     }
+
     const privateKey = await importJWK(key.privateKey, ED25519_ALG);
     const publicKey = await importJWK(key.publicKey, ED25519_ALG);
     if ('type' in privateKey && 'type' in publicKey) {
+      const publicKeyBase58 = TypedArrayEncoder.toBase58(
+        TypedArrayEncoder.fromBase64(
+          key.publicKey.x
+        )
+      );
+      const keyExists = await this.fetchKey(publicKeyBase58);
+
+      // if key already exists in the wallet
+      // returns keyId i.e. publicKeyBase58
+      if(keyExists != null) {
+        return {
+          keyId: publicKeyBase58,
+        };
+      }
+
       return await this.saveNewKey({ privateKey, publicKey });
     }
     throw new Error(`importJWK produced incorrect type. public key: ${publicKey}`);
   }
 
-  public async exportKey(keyDescription: KeyDescriptionDto): Promise<KeyPairDto> {
-    const keyPair = await this.keyRepository.findOneBy({ publicKeyThumbprint: keyDescription.keyId });
-    return keyPair;
+  public async exportKey(keyDescription: KeyDescriptionDto): Promise<Jwk> {
+    const key = await this.fetchKey(keyDescription.keyId);
+    return key?.key?.jwkPublic;
```

Comment:
Why only returning `jwkPublic` here? I think [jwkSecret](https://github.com/hyperledger/aries-askar/blob/781376830cceeaa80969dede01dd3ee969001f4e/wrappers/javascript/packages/aries-askar-shared/src/crypto/Key.ts#L91) is also available, which I assume is the privateKey JWK